### ISSUE TITLE
[1.3.0-RC1] No way to get around ${packaging.type}

### ISSUE THREAD

**Issue by anilkumarmyla** [sbt/sbt#4682]
First of all, great work on the 1.3.0-RC1, its super fast!

This might be a duplicate of https://github.com/sbt/sbt/issues/3618, but none of the workarounds described in there work for 1.3.0-RC1 release (with exception to excluding the dependency altogether).

Coursier claims to have solved this issue in https://github.com/coursier/coursier/issues/987, but it still creeps up in sbt-1.3.0-RC1 (using coursier underneath)

```
$ cat project/build.properties 
sbt.version=1.3.0-RC1
$ cat build.sbt 
libraryDependencies += "javax.ws.rs" % "javax.ws.rs-api" % "2.1.1"
$ sbt update
...
...
[error] java.net.URISyntaxException: Illegal character in path at index 88: https://repo1.maven.org/maven2/javax/ws/rs/javax.ws.rs-api/2.1.1/javax.ws.rs-api-2.1.1.${packaging.type}
[error] 	at java.base/java.net.URI$Parser.fail(URI.java:2936)
[error] 	at java.base/java.net.URI$Parser.checkChars(URI.java:3107)
[error] 	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3189)
[error] 	at java.base/java.net.URI$Parser.parse(URI.java:3137)
[error] 	at java.base/java.net.URI.<init>(URI.java:623)
[error] 	at java.base/java.net.URL.toURI(URL.java:1041)
[error] 	at sjsonnew.JavaExtraFormats.$anonfun$urlStringIso$1(JavaExtraFormats.scala:40)
[error] 	at sjsonnew.IsoString$$anon$1.to(IsoString.scala:26)
[error] 	at sjsonnew.IsoFormats$$anon$2.write(IsoFormats.scala:31)
[error] 	at sjsonnew.StandardFormats$OptionFormat.write(StandardFormats.scala:37)
[error] 	at sjsonnew.StandardFormats$OptionFormat.addField(StandardFormats.scala:44)
[error] 	at sjsonnew.StandardFormats$OptionFormat.addField(StandardFormats.scala:33)
[error] 	at sjsonnew.Builder.addField(Builder.scala:43)
[error] 	at sbt.librarymanagement.ArtifactFormats$$anon$1.write(ArtifactFormats.scala:35)
[error] 	at sbt.librarymanagement.ArtifactFormats$$anon$1.write(ArtifactFormats.scala:9)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error] 	at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] 	at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] 	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error] 	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error] 	at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error] 	at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error] 	at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error] 	at sjsonnew.Builder.addField(Builder.scala:43)
[error] 	at sbt.librarymanagement.ModuleReportFormats$$anon$1.write(ModuleReportFormats.scala:42)
[error] 	at sbt.librarymanagement.ModuleReportFormats$$anon$1.write(ModuleReportFormats.scala:9)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error] 	at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] 	at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] 	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error] 	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error] 	at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error] 	at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error] 	at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error] 	at sjsonnew.Builder.addField(Builder.scala:43)
[error] 	at sbt.librarymanagement.ConfigurationReportFormats$$anon$1.write(ConfigurationReportFormats.scala:26)
[error] 	at sbt.librarymanagement.ConfigurationReportFormats$$anon$1.write(ConfigurationReportFormats.scala:9)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error] 	at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] 	at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] 	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error] 	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error] 	at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error] 	at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error] 	at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error] 	at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error] 	at sjsonnew.Builder.addField(Builder.scala:43)
[error] 	at sbt.librarymanagement.UpdateReportFormats$$anon$1.write(UpdateReportFormats.scala:27)
[error] 	at sbt.librarymanagement.UpdateReportFormats$$anon$1.write(UpdateReportFormats.scala:9)
[error] 	at sjsonnew.SupportConverter.toJsonUnsafe(SupportConverter.scala:23)
[error] 	at sjsonnew.SupportConverter.toJsonUnsafe$(SupportConverter.scala:20)
[error] 	at sjsonnew.support.scalajson.unsafe.Converter$.toJsonUnsafe(Converter.scala:7)
[error] 	at sjsonnew.SupportConverter.$anonfun$toJson$1(SupportConverter.scala:14)
[error] 	at scala.util.Try$.apply(Try.scala:213)
[error] 	at sjsonnew.SupportConverter.toJson(SupportConverter.scala:14)
[error] 	at sjsonnew.SupportConverter.toJson$(SupportConverter.scala:13)
[error] 	at sjsonnew.support.scalajson.unsafe.Converter$.toJson(Converter.scala:7)
[error] 	at sbt.util.PlainOutput.write(Output.scala:16)
[error] 	at sbt.util.FileBasedStore.$anonfun$write$1(CacheStore.scala:73)
[error] 	at sbt.util.FileBasedStore.$anonfun$write$1$adapted(CacheStore.scala:72)
[error] 	at sbt.io.Using.apply(Using.scala:27)
[error] 	at sbt.util.FileBasedStore.write(CacheStore.scala:72)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:70)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:150)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:150)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:131)
[error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:150)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:164)
[error] 	at sbt.Classpaths$.$anonfun$updateTask0$5(Defaults.scala:2844)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:270)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:279)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:270)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:835)
[error] (update) java.net.URISyntaxException: Illegal character in path at index 88: https://repo1.maven.org/maven2/javax/ws/rs/javax.ws.rs-api/2.1.1/javax.ws.rs-api-2.1.1.${packaging.type}
```

**Comment by eed3si9n** [comment]
I just published the release note - https://www.lightbend.com/blog/sbt-1.3.0-release

Could you try

```scala
ThisBuild / useCoursier := false
```

to go back to Ivy, and see if it works again?

**Comment by anilkumarmyla** [comment]
It works with this
```
val PackagingTypeWorkaround: Unit = {
  sys.props += "packaging.type" -> "jar"
  ()
}

useCoursier := false
```

**Comment by eed3si9n** [comment]
ok. So as far as sbt 1.3.0 is concerned, the situation hasn't changed (if you use `useCouriser := false`) or we can ask Coursier to revisit the issue. Could we close this issue as sort of resolved? :)

**Comment by anilkumarmyla** [comment]
works for me, but disappointed I cannot use coursier :(

**Comment by eed3si9n** [comment]
If you have reproducible steps, you could open/reopen issue on Coursier repo.

**Comment by anilkumarmyla** [comment]
@eed3si9n I tried to reproduce this with coursier and interestingly, this resolves if I include the sbt-coursier plugin explicitly (anything going on in the background)?
```
$ cat project/build.properties 
sbt.version=1.3.0-RC1
$ cat project/plugins.sbt 
addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.1.0-M13-4")
$ cat build.sbt 
libraryDependencies += "javax.ws.rs" % "javax.ws.rs-api" % "2.1.1"
$ sbt update
```

**Comment by eed3si9n** [comment]
That's interesting. The core part of Coursier is the same but there are some glue code differences.

**Comment by vn971** [comment]
Possible duplicate of #956

**Comment by vn971** [comment]
This issue was raised in coursier as well (but just ultimately closed): https://github.com/coursier/coursier/issues/956

**Comment by anilkumarmyla** [comment]
This is still reproducible with 1.3.0-RC2 

Workaround is to add `addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.1.0-M14-4")`

But adding the sbt-coursier plugin leads to other unrelated failure when doing `compile` task
```
[error] java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
[error] 	at coursier.sbtcoursier.ResolutionTasks$.$anonfun$resolutionsTask$3(ResolutionTasks.scala:42)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:289)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:835)
[error] (coursierResolutions) java.lang.NoSuchMethodError: lmcoursier.definitions.ToCoursier$.project(Llmcoursier/definitions/Project;)Lcoursier/core/Project;
```

**Comment by antonkulaga** [comment]
I have same javax.ws.rs nightmare after update :(

**Comment by anilkumarmyla** [comment]
Good news, this problem resolves with the latest commit on master https://github.com/sbt/sbt/commit/0a54d8facce4ba3ece24f4283a11a0c7f4409f61

**Comment by eed3si9n** [comment]
I am closing this then.

**Comment by anilkumarmyla** [comment]
any more RC releases planned where we can test the same?

**Comment by eed3si9n** [comment]
RC-3 probably this week or next.

**Comment by anilkumarmyla** [comment]
can confirm this is fixed in 1.3.0-RC3 :woohoo:
